wagtail/tests/tests.py

```diff
@@ -443,38 +382,35 @@ def test_cache_clears_when_site_root_slug_changes(self):
         Fix: d6cce69a397d08d5ee81a8cbc1977ab2c9db2682
         Discussion: https://github.com/wagtail/wagtail/issues/157
         """
+        # Trigger cache population
+        Site.get_site_root_paths()
+
         # Get homepage
         homepage = Page.objects.get(url_path="/home/")
 
-        # Warm up the cache by getting the url
-        _ = homepage.url  # noqa
-
         # Change homepage title and slug
         homepage.title = "New home"
         homepage.slug = "new-home"
         homepage.save()
 
-        # Get fresh instance of homepage
-        homepage = Page.objects.get(id=homepage.id)
-
-        # Check url
-        self.assertEqual(homepage.url, "/")
+        # Check the cache was cleared
+        self.assertIsNone(Site._get_cached_site_root_paths())
 
     @override_settings(WAGTAIL_I18N_ENABLED=True)
     def test_cache_clears_when_site_root_is_translated_as_alias(self):
+        # Trigger cache population
+        Site.get_site_root_paths()
+
         # Get homepage
         homepage = Page.objects.get(url_path="/home/")
 
-        # Warm up the cache by getting the url
-        _ = homepage.url  # noqa
-
         # Translate the homepage
-        translated_homepage = homepage.copy_for_translation(
+        homepage.copy_for_translation(
             Locale.objects.create(language_code="fr"), alias=True
         )
 
-        # Check url
-        self.assertEqual(translated_homepage.url, "/")
```

Comment:
Question: Why are we no longer checking the URL?

Reply:
I've changed the nature of these tests subtly to be more unit-test like. The purpose of the test is now simply to assert that the cache is cleared. Whether or not the `url` method uses the cached values appropriately is a separate concern, and something to be tested elsewhere

Reply:
The docstrings for these tests have been simplified somewhat to set clearer expectations.